experimental/segmenter/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! A line breaker that is compatible with [Unicode Standard Annex #14][UAX14] and CSS properties.
+//!
+//! [UAX14]: http://www.unicode.org/reports/tr14/
+//!
+//!```rust
+//! use icu_segmenter::LineBreakIterator;
+//!
+//! fn main () {
+//!     let mut iter = LineBreakIterator::new("Hello World");
```

Comment:
Suggestion: The Intl.Segmenter proposal has two steps here: first creating the Segmenter object, which takes the locale and performs data loading, and then having a `.segment()` method on it that returns the iterator for the string.  I think a similar API would be good to have here as well, because when we implement dynamic data loading, we'll want to have a Segmenter object with that data loaded into it.

Reply:
Agree that we should have a top level API similar to the Intl.Segmenter proposal once we implement other types of breaker.
It's a bit strange to add one for `LineBreakIterator` because current Intl.Segmenter doesn't have `granularity: "line"`.

Reply:
I'm thinking more about the data loading. In your API, where does the segmentation data reside?

Reply:
In @Makotokato's current design, there are two sets of data: the rule table from UAX 14 (`UAX14_RULE_TABLE`) and the table converting codepoint to line break property (`UAX14_PROPERTY_TABLE`). They reside statically as arrarys in [rule_table.rs](https://github.com/unicode-org/icu4x/blob/4b8b5bd5d8137293f6647ffb0e2b4d2529d55d62/experimental/segmenter/src/rule_table.rs#L9) and [property_table.rs](https://github.com/unicode-org/icu4x/blob/4b8b5bd5d8137293f6647ffb0e2b4d2529d55d62/experimental/segmenter/src/property_table.rs#L10), respectively.

A use case of `UAX14_PROPERTY_TABLE` when converting a codepoint to line break property.
https://github.com/unicode-org/icu4x/blob/4b8b5bd5d8137293f6647ffb0e2b4d2529d55d62/experimental/segmenter/src/line_breaker.rs#L48

Reply:
Do you plan on making those arrays loadable through the data provider, in order to ensure we always have fresh segmentation data?

Reply:
Yes, loading the rule tables through the DataProvider is on my todo list. I'll investigate it after this PR is merged. There are more todos in "Possible future work after the integration" https://github.com/unicode-org/icu4x/issues/707#issue-878294852